### PR TITLE
New KS random metric

### PR DESCRIPTION
**Pull request by MaximArtemev**

### CODE REVIEW

File: coopetition/scoring_program/score.py [hunk 1]
```diff
@@ -33,8 +33,17 @@
     # prediction[config['X_cols']] = reference[config['X_cols']]
 
     score = 0
-    for col in config['Y_cols']:
-        score = max(score, ks_2samp(reference[col], prediction[col])[0])
+    kmax = 50
```
Comment (SiLiKhon) [inline coopetition/scoring_program/score.py]: Could you please rename `kmax` to something more meaningful? Maybe, `number_of_slices` or `n_slices`?
Also, you should read its value from `config`.

File: coopetition/config/config.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 training_file: "v1_pion_train.csv"
 scoring_file: "v1_pion_validation.csv"
+scoring_iter: 100
```
Comment (SiLiKhon) [inline coopetition/config/config.yml]: Not sure this is meaningful enough. I'd go with `number_of_slices`.

File: coopetition/scoring_program/score.py [hunk 3]
```diff
@@ -33,8 +33,17 @@
...
+    w_normal = np.random.normal(size=(kmax, len(config['Y_cols'])))
+    reference = reference.copy()[config['Y_cols']].values
+    prediction = prediction.copy()[config['Y_cols']].values
```
Comment (SiLiKhon) [inline coopetition/scoring_program/score.py]: Why are you making a copy here?

Comment (SiLiKhon) [reply coopetition/scoring_program/score.py]: Also, the metric should be done on a concatenation of X and Y vectors, e.g.

```python
XY_columns = config['X_cols'] + config['Y_cols']
reference[XY_columns]...
prediction[XY_columns]...
```

File: RICH_Data_Overview.ipynb [hunk 4]
```diff
@@ -1,575 +1,587 @@
...
...i0hhHcDuBjA7zf6KR8gtR1ERESAMVeUcTjDLcPUBfwSd/Pm+V7KOEOPM8U4W6/QQ85ZA7xcU66nOO5Z5p55ljlW2SJ+qZF7lls+kbfnzh0r23OfS/3na4tjPhdPBcidi+WmTMTnrncJytzj49ehsJwWT71hlAK7e/fxtMO9FP+AYt4/9k/JU9ezLOTg42ujELl9NULR3cbw+3c9gC8DuGHfHUPLMX4FwHIMrmhwn5mtDiHsW2/wY0PbO0FXth1yI5z1FIgFfAZDf/9siv3evb2+mGGc5c7FpP+J/uS+7W0+5nbO7rk+Ib/YzPEFIuMEBZ6JuZTGyzbTbIojTni9i3mWwgS6tvj4sxf6zLpXZvnvHZ658ZpZfjHHH27y6fo842H3bj915MQ5UXDaaX7ne/331BQqjJhrW/AUh0KRxjg74uyz3bZHjvGv4fGzfMuDp+ZyAgPNcMTD0WyOX6UL+frXfZxLEikurdo+Y/j7QqQl6ulQKKzhLCIiMua0cB5kCOFuM5tDj3sdgM0hhEcGL8e+AeACM/sJgCswuLpCSwshJqjtICIi0qHTJQ9EbpWHDRj+y98AHDPM/R1n3bpf0NJzftnIhx/2WQVnnVXrkuWe+wFfRwk4mYr6cDEcLgLEGQu8PX48ZzvwqHJunamS4h5z/lDwyDuP1HNvem4kP+4vz52L8XY+d24OT+r4ZYssslymRnw8qlqQnWOZOzZvj7MSCotr5Zbbog9+cRXJHoo5I+GR/bdC8OtI5ZeJ5AyF05HC+6umgrRF++dBHgsgroi3FcBpAN4H4GwAh5vZ3BDCVQ24xqwqtB1YfX8PLqd427B71TxKMS9v7f8Gx6UKbr3V70kD2li/Pn1mHnXesMHHM33JHLdKJTdL1qzxMTeRTqc/73yuww4b6SqB9ZQ595N/8duXL/fx6nt8RgKvnskZqBy/OKuWsXAIXxg9ken8xKnAI3+lcnugIEqfeGWpz0i49hN+18MP9xkJxx3nt19/vY/POcfHG6OJVPw0br7ZxwMD+s4UqctY6VAAcF5mu4iIiLTXcNMIQgjhSwC+1OqLgdoOIiIiY0ZulYcRS7Kb2Q8AvKHhV9RgS5ZMR19fvIzdGrd94kQ/WnvkkbXbvJoSz7F7ZbKfaziJH5Bbnoe761O9VDQnrzDKTKs88JE4n4FHwOMRbd43N3KfGy1PrZaQW2Ky7
```
Comment (SiLiKhon) [inline RICH_Data_Overview.ipynb]: Same comment as in score.py: you need to concatenate X and Y.

Comment (MaximArtemev) [reply RICH_Data_Overview.ipynb]: Yeah, but this is a sample notebook and for the sake of simplicity i guess we should left it as it was (with y only)

Comment (SiLiKhon) [reply RICH_Data_Overview.ipynb]: Since it introduces the students to the task, I'd prefer it to have the same metric as the competition.

**Pull request merged**